GooglePlayInstant/Tests/Editor/QuickDeploy/OAuth2ServerTest.cs

```diff
@@ -0,0 +1,109 @@
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using GooglePlayInstant.Editor.QuickDeploy;
+using NUnit.Framework;
+
+namespace GooglePlayInstant.Tests.Editor.QuickDeploy
+{
+    /// <summary>
+    /// Contains unit tests for OAuth2Server methods.
+    /// </summary>
+    [TestFixture]
+    public class OAuth2ServerTest
+    {
+        // Testing strategy:
+        //     - Test methods that do not require starting the server.
+        //     - TODO: Implement E2E tests to ensure that the server handles requests as expected.
+
+        private const string AddressPrefix = "http://localhost:5000/";
+
+        [Test]
+        public void TestEscapeAndUnescapeDataString()
+        {
+            Assert.AreEqual("%20", Uri.EscapeDataString(" "));
+            Assert.AreEqual(" ", Uri.UnescapeDataString("%20"));
+        }
+
+        [Test]
+        public void TestGetAuthorizationResponse()
+        {
+            var someString = Path.GetRandomFileName();
+            // response is code
+            var codeResponse =
+                OAuth2Server.GetAuthorizationResponse(new Uri(string.Format("{0}?code={1}", AddressPrefix,
+                    someString)));
+            Assert.AreEqual(new KeyValuePair<string, string>("code", someString), codeResponse,
+                "Expected valid code response");
+
+            // response is error
+            var errorResponse =
+                OAuth2Server.GetAuthorizationResponse(new Uri(string.Format("{0}?error={1}", AddressPrefix,
+                    someString)));
+            Assert.AreEqual(new KeyValuePair<string, string>("error", someString), errorResponse,
+                "Expected valid error response");
+
+            // response has invalid keys
+            var invalidResponse = new Uri(string.Format("{0}?someKey=someValue", AddressPrefix));
+            Assert.Throws<ArgumentException>(() => OAuth2Server.GetAuthorizationResponse(invalidResponse));
+
+            // No response. Uri has no params
+            Assert.Throws<ArgumentException>(() => OAuth2Server.GetAuthorizationResponse(new Uri(AddressPrefix)));
+        }
+
+        [Test]
+        public void TestGetQueryString()
+        {
+            var validUriWithCode = new Uri(string.Format("{0}?code=someValue", AddressPrefix));
+            Assert.AreEqual("code=someValue", OAuth2Server.GetQueryString(validUriWithCode));
+
+            var validUriWithError = new Uri(string.Format("{0}?error=someValue", AddressPrefix));
```

Comment:
GetQueryString doesn't care about the params. So this test is redundant and can be deleted.

Reply:
Resolved.